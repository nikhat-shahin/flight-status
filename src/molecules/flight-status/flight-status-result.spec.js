import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import { createSerializer } from 'enzyme-to-json';
import toJson from 'enzyme-to-json';
import mockedResponse from '../../api/flight-status.json';
import FlightStatus from './index';

configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

it('should render Page Component correctly', () => {
  const component = shallow(
    <FlightStatus flightStatusList={mockedResponse.results}/>,
  );
  expect(toJson(component)).toMatchSnapshot();
});

it('should render Page Component correctly for mobile view', () => {
  const component = shallow(
    <FlightStatus flightStatusList={mockedResponse.results} isMobile={true}/>,
  );
  expect(toJson(component)).toMatchSnapshot();
});



