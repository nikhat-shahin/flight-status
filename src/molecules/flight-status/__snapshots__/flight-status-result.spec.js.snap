// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Page Component correctly 1`] = `
<Container
  fluid={false}
>
  <Row
    className="flight-status__row1"
  >
    <Col
      className="flight-status__col"
    >
      <div
        className="flight-status__div"
      >
        <label
          className="flight-status__ddllabel"
        >
          Leaving from
        </label>
        <select
          className="flight-status__select"
          id="ddlLeavingFrom"
          onChange={[Function]}
          value="DXB"
        />
      </div>
    </Col>
    <Col
      className="flight-status__col"
    >
      <div
        className="flight-status__div"
      >
        <label
          className="flight-status__ddllabel"
        >
          Going to
        </label>
        <select
          className="flight-status__select"
          id="ddlGoingTo"
          onChange={[Function]}
          value="LHR"
        />
      </div>
    </Col>
    <Col
      className="flight-status__col"
    >
      <div
        className="flight-status__div"
      >
        <label
          className="flight-status__ddllabelDP"
        >
          Departing
        </label>
        <r
          allowSameDay={false}
          className="flight-status__dp"
          customTimeInput={null}
          dateFormat="dd MMM yyyy"
          dateFormatCalendar="LLLL yyyy"
          disabled={false}
          disabledKeyboardNavigation={false}
          dropdownMode="scroll"
          enableTabLoop={true}
          excludeScrollbar={true}
          focusSelectedMonth={false}
          monthsShown={1}
          name="TravelDate"
          nextMonthAriaLabel="Next Month"
          nextMonthButtonLabel="Next Month"
          nextYearAriaLabel="Next Year"
          nextYearButtonLabel="Next Year"
          onBlur={[Function]}
          onCalendarClose={[Function]}
          onCalendarOpen={[Function]}
          onChange={[Function]}
          onClickOutside={[Function]}
          onFocus={[Function]}
          onInputClick={[Function]}
          onInputError={[Function]}
          onKeyDown={[Function]}
          onMonthChange={[Function]}
          onSelect={[Function]}
          onYearChange={[Function]}
          preventOpenOnFocus={false}
          previousMonthAriaLabel="Previous Month"
          previousMonthButtonLabel="Previous Month"
          previousYearAriaLabel="Previous Year"
          previousYearButtonLabel="Previous Year"
          readOnly={false}
          renderDayContents={[Function]}
          selected={2022-03-21T18:24:11.540Z}
          selectsDisabledDaysInRange={false}
          shouldCloseOnSelect={true}
          showFourColumnMonthYearPicker={false}
          showFullMonthYearPicker={false}
          showMonthYearPicker={false}
          showPopperArrow={true}
          showPreviousMonths={false}
          showQuarterYearPicker={false}
          showTimeInput={false}
          showTimeSelect={false}
          showTwoColumnMonthYearPicker={false}
          showYearPicker={false}
          strictParsing={false}
          timeCaption="Time"
          timeInputLabel="Time"
          timeIntervals={30}
          withPortal={false}
          yearItemNumber={12}
        />
      </div>
    </Col>
    <Col
      className="flight-status__col"
    >
      <div
        className="flight-status__divbutton"
      >
        <button
          className="flight-status__button"
          onClick={[Function]}
        >
          View details
        </button>
      </div>
    </Col>
  </Row>
  <Row>
    <Col>
      <div
        className="flight-status"
      >
        <div
          key="flight-status_0"
        >
          <FlightStatus
            airlineDesignator="EK"
            flightDate="2022-03-16"
            flightId="2022031600007DXB"
            flightNumber="0007"
            flightRoute={
              Array [
                Object {
                  "arrivalTerminal": "Terminal 3",
                  "arrivalTime": Object {
                    "actual": "2022-03-17T07:08:00Z",
                    "estimated": "2022-03-17T06:53:00Z",
                    "schedule": "2022-03-17T07:10:00Z",
                  },
                  "departureTerminal": "Terminal 3",
                  "departureTime": Object {
                    "actual": "2022-03-17T03:15:00Z",
                    "estimated": "2022-03-17T03:10:00Z",
                    "schedule": "2022-03-17T03:10:00Z",
                  },
                  "destinationActualAirportCode": "London (LHR)",
                  "destinationPlannedAirportCode": "London (LHR)",
                  "flightOutageType": 0,
                  "flightPosition": 100,
                  "isIrregular": "false",
                  "legNumber": "1",
                  "operationalUpdate": Object {
                    "lastUpdated": "2022-01-20T04:00:55Z",
                  },
                  "originActualAirportCode": "Dubai (DXB)",
                  "originPlannedAirportCode": "Dubai (DXB)",
                  "statusCode": "ARVD",
                  "totalTravelDuration": "08:00",
                },
              ]
            }
          />
        </div>
        <div
          key="flight-status_1"
        >
          <FlightStatus
            airlineDesignator="EK"
            flightDate="2022-03-17"
            flightId="2022031700001DXB"
            flightNumber="0001"
            flightRoute={
              Array [
                Object {
                  "arrivalTerminal": "Terminal 3",
                  "arrivalTime": Object {
                    "actual": "2022-03-17T12:14:00Z",
                    "estimated": "2022-03-17T11:37:00Z",
                    "schedule": "2022-03-17T11:40:00Z",
                  },
                  "departureTerminal": "Terminal 3",
                  "departureTime": Object {
                    "actual": "2022-03-17T08:02:00Z",
                    "estimated": "2022-03-17T07:45:00Z",
                    "schedule": "2022-03-17T07:45:00Z",
                  },
                  "destinationActualAirportCode": "London (LHR)",
                  "destinationPlannedAirportCode": "London (LHR)",
                  "flightOutageType": 0,
                  "flightPosition": 100,
                  "isIrregular": "false",
                  "legNumber": "1",
                  "operationalUpdate": Object {
                    "lastUpdated": "2022-01-21T04:00:54Z",
                  },
                  "originActualAirportCode": "Dubai (DXB)",
                  "originPlannedAirportCode": "Dubai (DXB)",
                  "statusCode": "PDEP",
                  "totalTravelDuration": "07:55",
                },
              ]
            }
          />
        </div>
        <div
          key="flight-status_2"
        >
          <FlightStatus
            airlineDesignator="EK"
            flightDate="2022-03-17"
            flightId="2022031700029DXB"
            flightNumber="0029"
            flightRoute={
              Array [
                Object {
                  "arrivalTerminal": "Terminal 3",
                  "arrivalTime": Object {
                    "actual": "2022-03-17T13:28:00Z",
                    "estimated": "2022-03-17T13:27:00Z",
                    "schedule": "2022-03-17T13:50:00Z",
                  },
                  "departureTerminal": "Terminal 3",
                  "departureTime": Object {
                    "actual": "2022-03-17T09:52:00Z",
                    "estimated": "2022-03-17T09:40:00Z",
                    "schedule": "2022-03-17T09:40:00Z",
                  },
                  "destinationActualAirportCode": "London (LHR)",
                  "destinationPlannedAirportCode": "London (LHR)",
                  "flightOutageType": 0,
                  "flightPosition": 100,
                  "isIrregular": "false",
                  "legNumber": "1",
                  "operationalUpdate": Object {
                    "lastUpdated": "2022-01-21T04:00:54Z",
                  },
                  "originActualAirportCode": "Dubai (DXB)",
                  "originPlannedAirportCode": "Dubai (DXB)",
                  "statusCode": "ARVD",
                  "totalTravelDuration": "08:10",
                },
              ]
            }
          />
        </div>
        <div
          key="flight-status_3"
        >
          <FlightStatus
            airlineDesignator="EK"
            flightDate="2022-03-17"
            flightId="2022031700031DXB"
            flightNumber="0031"
            flightRoute={
              Array [
                Object {
                  "arrivalTerminal": "Terminal 3",
                  "arrivalTime": Object {
                    "actual": "2022-03-17T15:46:00Z",
                    "estimated": "2022-03-17T15:46:00Z",
                    "schedule": "2022-03-17T16:10:00Z",
                  },
                  "departureTerminal": "Terminal 3",
                  "departureTime": Object {
                    "actual": "2022-03-17T12:10:00Z",
                    "estimated": "2022-03-17T12:10:00Z",
                    "schedule": "2022-03-17T12:10:00Z",
                  },
                  "destinationActualAirportCode": "London (LHR)",
                  "destinationPlannedAirportCode": "London (LHR)",
                  "flightOutageType": 0,
                  "flightPosition": 100,
                  "isIrregular": "false",
                  "legNumber": "1",
                  "operationalUpdate": Object {
                    "lastUpdated": "2022-01-21T04:00:54Z",
                  },
                  "originActualAirportCode": "Dubai (DXB)",
                  "originPlannedAirportCode": "Dubai (DXB)",
                  "statusCode": "ARVD",
                  "totalTravelDuration": "08:00",
                },
              ]
            }
          />
        </div>
        <div
          key="flight-status_4"
        >
          <FlightStatus
            airlineDesignator="EK"
            flightDate="2022-03-17"
            flightId="2022031700003DXB"
            flightNumber="0003"
            flightRoute={
              Array [
                Object {
                  "arrivalTerminal": "Terminal 3",
                  "arrivalTime": Object {
                    "actual": "2022-03-17T18:46:00Z",
                    "estimated": "2022-03-17T18:45:00Z",
                    "schedule": "2022-03-17T18:20:00Z",
                  },
                  "departureTerminal": "Terminal 3",
                  "departureTime": Object {
                    "actual": "2022-03-17T14:59:00Z",
                    "estimated": "2022-03-17T15:10:00Z",
                    "schedule": "2022-03-17T14:30:00Z",
                  },
                  "destinationActualAirportCode": "London (LHR)",
                  "destinationPlannedAirportCode": "London (LHR)",
                  "flightOutageType": 0,
                  "flightPosition": 100,
                  "isIrregular": "false",
                  "legNumber": "1",
                  "operationalUpdate": Object {
                    "lastUpdated": "2022-01-21T04:00:54Z",
                  },
                  "originActualAirportCode": "Dubai (DXB)",
                  "originPlannedAirportCode": "Dubai (DXB)",
                  "statusCode": "ARVD",
                  "totalTravelDuration": "07:50",
                },
              ]
            }
          />
        </div>
        <div
          key="flight-status_5"
        >
          <FlightStatus
            airlineDesignator="EK"
            flightDate="2022-03-17"
            flightId="2022031700005DXB"
            flightNumber="0005"
            flightRoute={
              Array [
                Object {
                  "arrivalTerminal": "Terminal 3",
                  "arrivalTime": Object {
                    "actual": "2022-03-17T19:51:00Z",
                    "estimated": "2022-03-17T19:52:00Z",
                    "schedule": "2022-03-17T20:00:00Z",
                  },
                  "departureTerminal": "Terminal 3",
                  "departureTime": Object {
                    "actual": "2022-03-17T16:23:00Z",
                    "estimated": "2022-03-17T16:05:00Z",
                    "schedule": "2022-03-17T16:05:00Z",
                  },
                  "destinationActualAirportCode": "London (LHR)",
                  "destinationPlannedAirportCode": "London (LHR)",
                  "flightOutageType": 0,
                  "flightPosition": 100,
                  "isIrregular": "false",
                  "legNumber": "1",
                  "operationalUpdate": Object {
                    "lastUpdated": "2022-01-21T04:00:54Z",
                  },
                  "originActualAirportCode": "Dubai (DXB)",
                  "originPlannedAirportCode": "Dubai (DXB)",
                  "statusCode": "ARVD",
                  "totalTravelDuration": "07:55",
                },
              ]
            }
          />
        </div>
      </div>
    </Col>
  </Row>
</Container>
`;

exports[`should render Page Component correctly for mobile view 1`] = `
<Container
  fluid={false}
>
  <Row
    className="flight-status__row1"
  >
    <Col
      className="flight-status__col"
    >
      <div
        className="flight-status__div"
      >
        <label
          className="flight-status__ddllabel"
        >
          Leaving from
        </label>
        <select
          className="flight-status__select"
          id="ddlLeavingFrom"
          onChange={[Function]}
          value="DXB"
        />
      </div>
    </Col>
    <Col
      className="flight-status__col"
    >
      <div
        className="flight-status__div"
      >
        <label
          className="flight-status__ddllabel"
        >
          Going to
        </label>
        <select
          className="flight-status__select"
          id="ddlGoingTo"
          onChange={[Function]}
          value="LHR"
        />
      </div>
    </Col>
    <Col
      className="flight-status__col"
    >
      <div
        className="flight-status__div"
      >
        <label
          className="flight-status__ddllabelDP"
        >
          Departing
        </label>
        <r
          allowSameDay={false}
          className="flight-status__dp"
          customTimeInput={null}
          dateFormat="dd MMM yyyy"
          dateFormatCalendar="LLLL yyyy"
          disabled={false}
          disabledKeyboardNavigation={false}
          dropdownMode="scroll"
          enableTabLoop={true}
          excludeScrollbar={true}
          focusSelectedMonth={false}
          monthsShown={1}
          name="TravelDate"
          nextMonthAriaLabel="Next Month"
          nextMonthButtonLabel="Next Month"
          nextYearAriaLabel="Next Year"
          nextYearButtonLabel="Next Year"
          onBlur={[Function]}
          onCalendarClose={[Function]}
          onCalendarOpen={[Function]}
          onChange={[Function]}
          onClickOutside={[Function]}
          onFocus={[Function]}
          onInputClick={[Function]}
          onInputError={[Function]}
          onKeyDown={[Function]}
          onMonthChange={[Function]}
          onSelect={[Function]}
          onYearChange={[Function]}
          preventOpenOnFocus={false}
          previousMonthAriaLabel="Previous Month"
          previousMonthButtonLabel="Previous Month"
          previousYearAriaLabel="Previous Year"
          previousYearButtonLabel="Previous Year"
          readOnly={false}
          renderDayContents={[Function]}
          selected={2022-03-21T18:24:11.816Z}
          selectsDisabledDaysInRange={false}
          shouldCloseOnSelect={true}
          showFourColumnMonthYearPicker={false}
          showFullMonthYearPicker={false}
          showMonthYearPicker={false}
          showPopperArrow={true}
          showPreviousMonths={false}
          showQuarterYearPicker={false}
          showTimeInput={false}
          showTimeSelect={false}
          showTwoColumnMonthYearPicker={false}
          showYearPicker={false}
          strictParsing={false}
          timeCaption="Time"
          timeInputLabel="Time"
          timeIntervals={30}
          withPortal={false}
          yearItemNumber={12}
        />
      </div>
    </Col>
    <Col
      className="flight-status__col"
    >
      <div
        className="flight-status__divbutton"
      >
        <button
          className="flight-status__button"
          onClick={[Function]}
        >
          View details
        </button>
      </div>
    </Col>
  </Row>
  <Row>
    <Col>
      <div
        className="flight-status"
      >
        <div
          key="flight-status_0"
        >
          <FlightStatus
            airlineDesignator="EK"
            flightDate="2022-03-16"
            flightId="2022031600007DXB"
            flightNumber="0007"
            flightRoute={
              Array [
                Object {
                  "arrivalTerminal": "Terminal 3",
                  "arrivalTime": Object {
                    "actual": "2022-03-17T07:08:00Z",
                    "estimated": "2022-03-17T06:53:00Z",
                    "schedule": "2022-03-17T07:10:00Z",
                  },
                  "departureTerminal": "Terminal 3",
                  "departureTime": Object {
                    "actual": "2022-03-17T03:15:00Z",
                    "estimated": "2022-03-17T03:10:00Z",
                    "schedule": "2022-03-17T03:10:00Z",
                  },
                  "destinationActualAirportCode": "London (LHR)",
                  "destinationPlannedAirportCode": "London (LHR)",
                  "flightOutageType": 0,
                  "flightPosition": 100,
                  "isIrregular": "false",
                  "legNumber": "1",
                  "operationalUpdate": Object {
                    "lastUpdated": "2022-01-20T04:00:55Z",
                  },
                  "originActualAirportCode": "Dubai (DXB)",
                  "originPlannedAirportCode": "Dubai (DXB)",
                  "statusCode": "ARVD",
                  "totalTravelDuration": "08:00",
                },
              ]
            }
          />
        </div>
        <div
          key="flight-status_1"
        >
          <FlightStatus
            airlineDesignator="EK"
            flightDate="2022-03-17"
            flightId="2022031700001DXB"
            flightNumber="0001"
            flightRoute={
              Array [
                Object {
                  "arrivalTerminal": "Terminal 3",
                  "arrivalTime": Object {
                    "actual": "2022-03-17T12:14:00Z",
                    "estimated": "2022-03-17T11:37:00Z",
                    "schedule": "2022-03-17T11:40:00Z",
                  },
                  "departureTerminal": "Terminal 3",
                  "departureTime": Object {
                    "actual": "2022-03-17T08:02:00Z",
                    "estimated": "2022-03-17T07:45:00Z",
                    "schedule": "2022-03-17T07:45:00Z",
                  },
                  "destinationActualAirportCode": "London (LHR)",
                  "destinationPlannedAirportCode": "London (LHR)",
                  "flightOutageType": 0,
                  "flightPosition": 100,
                  "isIrregular": "false",
                  "legNumber": "1",
                  "operationalUpdate": Object {
                    "lastUpdated": "2022-01-21T04:00:54Z",
                  },
                  "originActualAirportCode": "Dubai (DXB)",
                  "originPlannedAirportCode": "Dubai (DXB)",
                  "statusCode": "PDEP",
                  "totalTravelDuration": "07:55",
                },
              ]
            }
          />
        </div>
        <div
          key="flight-status_2"
        >
          <FlightStatus
            airlineDesignator="EK"
            flightDate="2022-03-17"
            flightId="2022031700029DXB"
            flightNumber="0029"
            flightRoute={
              Array [
                Object {
                  "arrivalTerminal": "Terminal 3",
                  "arrivalTime": Object {
                    "actual": "2022-03-17T13:28:00Z",
                    "estimated": "2022-03-17T13:27:00Z",
                    "schedule": "2022-03-17T13:50:00Z",
                  },
                  "departureTerminal": "Terminal 3",
                  "departureTime": Object {
                    "actual": "2022-03-17T09:52:00Z",
                    "estimated": "2022-03-17T09:40:00Z",
                    "schedule": "2022-03-17T09:40:00Z",
                  },
                  "destinationActualAirportCode": "London (LHR)",
                  "destinationPlannedAirportCode": "London (LHR)",
                  "flightOutageType": 0,
                  "flightPosition": 100,
                  "isIrregular": "false",
                  "legNumber": "1",
                  "operationalUpdate": Object {
                    "lastUpdated": "2022-01-21T04:00:54Z",
                  },
                  "originActualAirportCode": "Dubai (DXB)",
                  "originPlannedAirportCode": "Dubai (DXB)",
                  "statusCode": "ARVD",
                  "totalTravelDuration": "08:10",
                },
              ]
            }
          />
        </div>
        <div
          key="flight-status_3"
        >
          <FlightStatus
            airlineDesignator="EK"
            flightDate="2022-03-17"
            flightId="2022031700031DXB"
            flightNumber="0031"
            flightRoute={
              Array [
                Object {
                  "arrivalTerminal": "Terminal 3",
                  "arrivalTime": Object {
                    "actual": "2022-03-17T15:46:00Z",
                    "estimated": "2022-03-17T15:46:00Z",
                    "schedule": "2022-03-17T16:10:00Z",
                  },
                  "departureTerminal": "Terminal 3",
                  "departureTime": Object {
                    "actual": "2022-03-17T12:10:00Z",
                    "estimated": "2022-03-17T12:10:00Z",
                    "schedule": "2022-03-17T12:10:00Z",
                  },
                  "destinationActualAirportCode": "London (LHR)",
                  "destinationPlannedAirportCode": "London (LHR)",
                  "flightOutageType": 0,
                  "flightPosition": 100,
                  "isIrregular": "false",
                  "legNumber": "1",
                  "operationalUpdate": Object {
                    "lastUpdated": "2022-01-21T04:00:54Z",
                  },
                  "originActualAirportCode": "Dubai (DXB)",
                  "originPlannedAirportCode": "Dubai (DXB)",
                  "statusCode": "ARVD",
                  "totalTravelDuration": "08:00",
                },
              ]
            }
          />
        </div>
        <div
          key="flight-status_4"
        >
          <FlightStatus
            airlineDesignator="EK"
            flightDate="2022-03-17"
            flightId="2022031700003DXB"
            flightNumber="0003"
            flightRoute={
              Array [
                Object {
                  "arrivalTerminal": "Terminal 3",
                  "arrivalTime": Object {
                    "actual": "2022-03-17T18:46:00Z",
                    "estimated": "2022-03-17T18:45:00Z",
                    "schedule": "2022-03-17T18:20:00Z",
                  },
                  "departureTerminal": "Terminal 3",
                  "departureTime": Object {
                    "actual": "2022-03-17T14:59:00Z",
                    "estimated": "2022-03-17T15:10:00Z",
                    "schedule": "2022-03-17T14:30:00Z",
                  },
                  "destinationActualAirportCode": "London (LHR)",
                  "destinationPlannedAirportCode": "London (LHR)",
                  "flightOutageType": 0,
                  "flightPosition": 100,
                  "isIrregular": "false",
                  "legNumber": "1",
                  "operationalUpdate": Object {
                    "lastUpdated": "2022-01-21T04:00:54Z",
                  },
                  "originActualAirportCode": "Dubai (DXB)",
                  "originPlannedAirportCode": "Dubai (DXB)",
                  "statusCode": "ARVD",
                  "totalTravelDuration": "07:50",
                },
              ]
            }
          />
        </div>
        <div
          key="flight-status_5"
        >
          <FlightStatus
            airlineDesignator="EK"
            flightDate="2022-03-17"
            flightId="2022031700005DXB"
            flightNumber="0005"
            flightRoute={
              Array [
                Object {
                  "arrivalTerminal": "Terminal 3",
                  "arrivalTime": Object {
                    "actual": "2022-03-17T19:51:00Z",
                    "estimated": "2022-03-17T19:52:00Z",
                    "schedule": "2022-03-17T20:00:00Z",
                  },
                  "departureTerminal": "Terminal 3",
                  "departureTime": Object {
                    "actual": "2022-03-17T16:23:00Z",
                    "estimated": "2022-03-17T16:05:00Z",
                    "schedule": "2022-03-17T16:05:00Z",
                  },
                  "destinationActualAirportCode": "London (LHR)",
                  "destinationPlannedAirportCode": "London (LHR)",
                  "flightOutageType": 0,
                  "flightPosition": 100,
                  "isIrregular": "false",
                  "legNumber": "1",
                  "operationalUpdate": Object {
                    "lastUpdated": "2022-01-21T04:00:54Z",
                  },
                  "originActualAirportCode": "Dubai (DXB)",
                  "originPlannedAirportCode": "Dubai (DXB)",
                  "statusCode": "ARVD",
                  "totalTravelDuration": "07:55",
                },
              ]
            }
          />
        </div>
      </div>
    </Col>
  </Row>
</Container>
`;
